### DISCUSSION TITLE
Should the webdav access for originals folder be read-only?

### DISCUSSION THREAD

**Discussion by OSHW-Rico** [photoprism/photoprism#1844] [Q&A] [answered]
I just tried the webdav access and was "shocked" that the originals folder is NOT read only. It is possible to delete originals. This makes no sense in my opinion, why would someone want to do something like that? Photoprism will not be aware of changes or removed photos. You will only recognise on the day you need the original.

The Download will then fail "Failed - No File"

Am I missing something?

**Comment by lastzero** [comment, marked as answer]
Yes, PhotoPrism automatically updates the index when you delete files via WebDAV. Why is this shocking?

**Reply by OSHW-Rico** [reply]
Okay, i will check again after the Import of this folder is done if the index got updated.

Shocking is a bit harsh, unexpected. The intention was to get easy access to the originals without the possibility to remove originals. See the Docs regarding the originals folder:

https://docs.photoprism.app/user-guide/advanced/folders/#originals

`PhotoPrism will never move or rename files in this folder`

Delete is more extreme than move or rename. That is why I was not expecting it to be be able to delete via WebDAV. At least not in a default setting.

**Reply by lastzero** [reply]
You deleted the file, not photoprism. We will make this even more clear to prevent further shocks.

**Reply by OSHW-Rico** [reply]
True :100: 

I will mount the originals folder via external webDAV as readonly.

**Reply by lastzero** [reply]
Don't panic 👍

**Reply by jpbuttet** [reply]
> True 💯
> 
> I will mount the originals folder via external webDAV as readonly.

I have such config, i put following line in my host's fstab:
https://xxx.xxx.xxx.xx/myserver/webdav /media/photoprismwebdav davfs ro,user,noauto 0 0

and I added following line in photoprism docker-compose.yml :
- "/media/photoprismwebdav:/photoprism/originals"


it works fine but then it raised a new question:
What happens if the external webDAV is not mounted before photoprism container startup ?
In this case, I think that the original folder would appear as empty to photoprism and photoprism would automatically start to index that empty folder and depopulate the database.
What do you think ?
Is it possible to disable protoprism indexing if the originals folder is empty ?
Or is it better to check that the originals folder is not empty (or external webdav mounted) in the docker-compose.yml file ?

Best regards.

**Reply by lastzero** [reply]
> In this case, I think that the original folder would appear as empty to photoprism and photoprism would automatically start to index that empty folder and depopulate the database.

Yes, that would currently be the case unless we add (yet another) config option to change it...

**Reply by jpbuttet** [reply]
Thank you for confirming that point.
